Update go version in go.mod to fix issue #10

juandspy opened the pull request:
### Context:
Updating the go version from 1.15 to 1.18 in order to fix issue #10.

### Testing steps:
I created a Dockerfile:
```Dockerfile
FROM golang:1.18

COPY . /app

WORKDIR /app

RUN make install
```
And then run:
```
❯ docker build -t terraform-provider-rhoas .
```

jackdelahunt reviewed (approved):
Verify steps worked just need to fix the linting errors

Run `make lint`

juandspy commented:
> Verify steps worked just need to fix the linting errors
> 
> Run `make lint`

@jackdelahunt Interesting. I was using go1.18 locally and with that version the linting passes. However, with go1.19 it doesn't. Updated and fixed the linting! :slightly_smiling_face:

jackdelahunt commented:
Welcome to the world of linters that sometimes work and don't. Just stay on the same version as the CI one and you are good 👍
